File: app/src/main/java/com/ntrllog/bagels/ExitDialogFragment.java
package com.ntrllog.bagels;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

public class ExitDialogFragment extends DialogFragment {

    private Context context;

    public ExitDialogFragment(Context c) {
        this.context = c;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage(R.string.exit_message)
                .setPositiveButton(R.string.exit, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        Intent intent = new Intent(context, MainActivity.class);
                        startActivity(intent);
                    }
                })
                .setNegativeButton(R.string.stay, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                    }
                });
        return builder.create();
    }
}

File: app/src/main/java/ntrllog/github/io/bagels/ExitDialogFragment.kt
package ntrllog.github.io.bagels

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.DialogFragment

class ExitDialogFragment(private val context: Context) : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setMessage(R.string.exit_message)
            .setPositiveButton(R.string.exit) { dialog, id ->
                val intent = Intent(context, MainActivity::class.java)
                startActivity(intent)
            }
            .setNegativeButton(R.string.stay) { dialog, id -> }
        return builder.create()
    }
}
